### DISCUSSION TITLE
Adding "inverse relationship" computed properties to models

### DISCUSSION THREAD

**Discussion by JHacin** [payloadcms/payload#967] [Q&A]
Hey everyone,

been trying out Payload CMS for the last few days, looking good so far. However, I'm not sure how to approach this requirement I have. Disclaimer: the issue might just be that I'm more used to relational DBs.

Let's say we have 3 models:

```
USER
_id
email

PRODUCT
_id
sku

ORDER
_id
user_id
product_id
```

The requirement: whenever I query a `User` document (whether I query by ID or the entire collection), I would also like to include an `orders` property, which is an array of references to the orders that the queried `User` is part of.

Possible solutions that come to mind right now:
- using some sort of collection hook like `afterRead` in the `User` config - causes an infinite loop when querying the `Order` collection unless you use `depth: 0` in `payload.find()`
- using custom API endpoints (would have to maintain at least two separate ones though (GET all + GET by id))
- adding a hidden `orders` property to User and manually updating it whenever the `user_id` property changes on any `Order` document

Am I missing an obvious solution here?

Thanks.

**Comment by powerpaul17** [comment]
I think there is a join field for that
